### DISCUSSION TITLE
Dynamically build the chain networks based on a backend (axios) response

### DISCUSSION THREAD

**Discussion by dcalap** [rainbow-me/rainbowkit#1184] [Q&A]
I want to dynamically build the chains/networks that will appear in the button based on a configuration that I will get from a backend by using axios.

I have the wagmi.ts file where I configure the chains:

const { chains, provider, webSocketProvider } = configureChains(
  defaultChains,
    [publicProvider()],
)

Is this the way to go to provide a defaultChains array based on my response? If yes, what is the way to do it? I tried and the app fails, telling me that can't use a hook there (I've tried with useEffect for calling the backend there) in that file. I'm not an expert in React. 

Thanks!

**Comment by DanielSinclair** [comment]
This is typically not recommended because of Wagmi's architecture. But if you're usecase is that different pages support different chains, you should be able to supply your full chain list to Wagmi, and then offer a subset of the chain list to the `RainbowKitProvider`'s `chains` prop to show that subset of chains to the user. You would want to create a provider wrapper to do that cleanly.
